lib/ood_core/job/adapters/slurm.rb

```diff
@@ -617,6 +617,12 @@ def get_state(st)
             STATE_MAP.fetch(st, :undetermined)
           end
 
+          def get_gpu_count(gres)
+            gpus = 0
+            gres.to_s.scan(/gpu:[^,]*(\d+)/).each { |val| gpus += val[0].to_i }
+            gpus
```

Comment:
```suggestion
            gres.to_s.scan(/gpu:[^,]*(\d+)/).flatten.map(&:to_i).sum
```

Anytime you find yourself doing something like this - initializing outside a loop then returning that - in Ruby there's likely a much better way.

Reply:
Specifically like `inject` or `each_with_object`, but in this case, we can directly just remap and sum the array.